clang/lib/AST/Decl.cpp

```diff
@@ -2434,6 +2434,22 @@ VarDecl *VarDecl::getInitializingDeclaration() {
   return Def;
 }
 
+bool VarDecl::hasInitWithSideEffects() const {
+  if (!hasInit())
+    return false;
+
+  if (auto *S = dyn_cast<Stmt *>(Init)) {
```

Comment:
The original code in `getInit` also handles `EvaluatedStmt`. I would have naively expected we would call `getEvaluatedStmt()` and if present, we would check if it's an offset (in which case fall through to asking the `ASTReader::hasInitializerWithSideEffects`) or a pointer, in which case handle it here.  But I'm not that familiar with `EvaluatedStmt` so maybe I missed something

Reply:
I'm not sure if it matters either, but it makes sense to use the actual expression as the source of truth when possible